src/container_registry/ecr.rs

```diff
@@ -263,6 +281,82 @@ impl ECR {
 
         self.create_repository(image)
     }
+
+    fn get_credentials(&self) -> Result<ECRCredentials, EngineError> {
+        let r = block_on(
+            self.ecr_client()
+                .get_authorization_token(GetAuthorizationTokenRequest::default()),
+        );
+
+        let (access_token, password, endpoint_url) = match r {
+            Ok(t) => match t.authorization_data {
+                Some(authorization_data) => {
+                    let ad = authorization_data.first().unwrap();
+                    let b64_token = ad.authorization_token.as_ref().unwrap();
+
+                    let decoded_token = base64::decode(b64_token).unwrap();
+                    let token = std::str::from_utf8(decoded_token.as_slice()).unwrap();
+
+                    let s_token: Vec<&str> = token.split(':').collect::<Vec<_>>();
+
+                    (
+                        s_token.first().unwrap().to_string(),
+                        s_token.get(1).unwrap().to_string(),
+                        ad.clone().proxy_endpoint.unwrap(),
+                    )
+                }
+                None => {
+                    return Err(self.engine_error(
+                        EngineErrorCause::Internal,
+                        format!(
+                            "failed to retrieve credentials and endpoint URL from ECR {}",
+                            self.name_with_id(),
+                        ),
+                    ));
+                }
+            },
+            _ => {
+                return Err(self.engine_error(
+                    EngineErrorCause::Internal,
+                    format!(
+                        "failed to retrieve credentials and endpoint URL from ECR {}",
+                        self.name_with_id(),
+                    ),
+                ));
+            }
+        };
+
+        Ok(ECRCredentials::new(access_token, password, endpoint_url))
+    }
+
+    fn exec_docker_login(&self) -> Result<(), EngineError> {
```

Comment:
Maybe worth moving this one into a common docker place. It's already used on a very similar way on SCW and DO already. Can be done later on though.

Reply:
yes agreed

Reply:
Hum, after taking a look, I am sure it's done the same way for DockerHub, DOCR, and SCW Registry. For ECR we add the `endpoint URL` at the end of the `docker login` command. We can factorize it, but I wonder if it is really useful.

Reply:
Ok !